middleware/multipart.js

```diff
@@ -91,10 +95,37 @@ module.exports = function (config) {
                 return;
             }
 
-            req.body = fields;
-            req.files = files;
-            res.once('finish', cleanify(files));
+            // Use firstValues to handle formidable 3.x array format
+            req.body = firstValues(form, fields);
+            req.files = addBackwardCompatibility(firstValues(form, files));
+            res.once('finish', cleanify(req.files));
             next();
         });
     });
-};
\ No newline at end of file
+};
+
+
+/**
+ * Add backward compatibility properties to file objects
+ * @param files the files object from formidable
+ */
+function addBackwardCompatibility(files) {
+    if (!files) {
+        return {};
+    }
+    
+    Object.keys(files).forEach(function(key) {
+        var file = files[key];
+        if (file) {
+            // Add backward compatibility properties for both formidable versions
```

Comment:
What's the purpose of shimming the previous version? It would be better to operate under the pretense of the new API and shim the old version into the new shape. Then when it's time to drop the old, we just drop.the shim portion.